### ISSUE TITLE
Incorrect method signature called from graal-js

### ISSUE THREAD

**Issue by janglada** [oracle/graal#3022]
**Describe GraalVM and your environment :**
 - GraalVM version or commit id if built from source: **20.3**
 - CE or EE: **CE**
 - JDK version: **JDK11**
 - OS and OS Version: **[macOS Mojave]**
 - Architecture: **amd64**
 - The output of `java -Xinternalversion`: ```OpenJDK 64-Bit Server VM (11.0.9+10-jvmci-20.3-b06) for bsd-amd64 JRE (11.0.9+10-jvmci-20.3-b06), built on Nov 11 2020 22:26:48 by "graal1" with gcc 4.2.1 Compatible Apple LLVM 10.0.0 (clang-1000.11.45.5)```

**Describe the issue**
When invoking a method with multiple signatures from JS with vargargs, the executed method is not the best candidate.

**Code snippet or code repository that reproduces the issue**
```
public class TestVarArgs {

	/**
	 *
	 */
	public static class MyObject {
		public String execute(String name, Function<Integer, String> func, Object ...args) {
			Assertions.assertNotNull(func);
			Assertions.assertEquals(3, args.length);

			return func.apply(1);
		}
	}

	/**
	 *
	 */
	public static class MyObject2 {

                // this signature should not be invoked
		public String execute(String name, Object ...args) {
			return execute(name, null, args);
		}

		public String execute(String name, Function<Integer, String> func, Object ...args) {
			Assertions.assertNotNull(func);
			Assertions.assertEquals(3, args.length);
			return func.apply(1);
		}
	}

	/**
	 *
	 * @throws ScriptException
	 */
	@Test
	public void success()
		throws ScriptException
	{
		ScriptEngine engine = getEngine();
		engine.put("myobject", new MyObject());
		engine.eval("myobject.execute('a', (idx) => idx  == 0? 'a': 'b', 1,2,3)");
	}


	/**
	 *
	 * @throws ScriptException
	 */
	@Test
	public void error()
		throws ScriptException
	{
		ScriptEngine engine = getEngine();
		engine.put("myobject", new MyObject2());
		engine.eval("myobject.execute('a', (idx) => idx  == 0? 'a': 'b', 1,2,3)");
	}


	/**
	 *
	 * @return
	 */
	private ScriptEngine getEngine() {
		ScriptEngine engine = new ScriptEngineManager().getEngineByName("Graal.js");
		engine.put("polyglot.js.allowHostAccess", true);
		return engine;
	}

}
```

**Steps to reproduce the issue**

1. Step one [e.g.: git clone  https://github.com/janglada/sample-graal.git ]
2. Step two [e.g.: ./gradlew test]

**Expected behavior**

When invoking ``execute`` on ``MyObject2`` I would expect the method with the following signature to be called:
```
public String execute(String name, Function<Integer, String> func, Object ...args)
```

instead the following method is invoked:

```
public String execute(String name,  Object ...args)
```
which cause args

**Comment by mcraj017** [comment]
@janglada thanks for reporting the issue
We will check it and update you

**Comment by woess** [comment]
Fixed in 21.0.0 (https://github.com/oracle/graal/commit/5058707b0a95fa14c6f2e2f89e7de3da0a302d2c). The overload with the `Function` parameter is now preferred. Thanks again for the report.
